extensions/src/main/java/org/pentaho/platform/web/http/api/resources/RepositoryResource.java

```diff
@@ -650,6 +654,23 @@ private boolean hasOperationId( final List<IPluginOperation> operations, final S
 
   protected Response doService( String contextId, String resourceId ) throws ObjectFactoryException,
       PluginBeanException, IOException, URISyntaxException {
+    // Check if we are editing a content
+    String perspectiveId = resourceId;
+    if ( perspectiveId != null && perspectiveId.indexOf( "." ) >= 0 ) {
+      String[] parts = perspectiveId.split( "\\." );
+      if( parts != null && parts.length > 0) {
+        perspectiveId = parts[1];
+      }
+    }
+
+    if( perspectiveId != null && ( perspectiveId.equals( "editor" ) || perspectiveId.equals( "edit" ) ) ) {
+      // Check if user has permission to edit the content. If they do not have access, throw and error
+      if( !canEdit() ) {
```

Comment:
also unit test for this method. should be similar to `FileServiceTest#testDoCanEdit`

Reply:
I wish we have good testing coverage for all resource classes. Since we were not able to accomplish that we split the code into resource and service, where we can test the service code executing the logic. We do not have unit test for this method at this moment